### PR TITLE
Fixes #7, #10

### LINKED ISSUE #7: Webhook can be triggered with other request methods than POST
I was testing this plugin and noticed that I can trigger the webhook with other types of request methods than POST, for example with GET and HEAD requests.

```bash
% curl --verbose -X GET http://localhost:8080/webhook-step/ba0860a4-63bd-4377-ac5a-45de0d332bc9
Note: Unnecessary use of -X or --request, GET is already inferred.
...
> GET /webhook-step/ba0860a4-63bd-4377-ac5a-45de0d332bc9 HTTP/1.1
> Host: localhost
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Tue, 13 Mar 2018 09:31:35 GMT
< Expires: Thu, 01 Jan 1970 00:00:00 GMT
< Result: WebhookTriggered
< Server: nginx
< Set-Cookie: JSESSIONID.37149ec0=node04n366t060xd7d28o78glcu0y482.node0;Path=/;HttpOnly
< Strict-Transport-Security: max-age=10886400; includeSubDomains; preload
< X-Content-Type-Options: nosniff
< Content-Length: 0
< Connection: keep-alive
```

This is in the Jenkins job:
```
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] registerWebhook
[Pipeline] echo
Waiting for POST to http://localhost:8080/webhook-step/ba0860a4-63bd-4377-ac5a-45de0d332bc9
[Pipeline] waitForWebhook
[Pipeline] echo
Webhook called with data: 
[Pipeline] End of Pipeline
Finished: SUCCESS
```

### LINKED ISSUE #10: Access to request headers
It would be super useful if the plugin gave additionally access to request headers as they can contain details regarding the triggered long running job status.

One such use case would be asynchronous invoking of [openfaas](https://github.com/openfaas/faas) functions - they would call back the webhook with following headers:

```
x-function-status: 200
x-duration-seconds: 60.130061
x-call-id: 123456789 
```
Cheers!

### PR DESCRIPTION
**Pull request by slide**
- Adds parameter to the wait step to request content and headers be returned.
- Added some tests for various issues that have popped up in the past

### CODE REVIEW

**Comment by slide** [comment]
@jenkinsci/code-reviewers Would like some feedback on this fix. My main concern is any possible security implications in exposing the headers to a pipeline script. Part of me says it shouldn't be a concern because the incoming payload shouldn't be secure, but I wanted some external eyes looking at it.

File: pom.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
   <properties>
     <jenkins.version>2.138.4</jenkins.version>
     <java.level>8</java.level>
+    <workflow-support.version>2.13</workflow-support.version>
```
Comment (jglick) [inline pom.xml]: consider using https://github.com/jenkinsci/bom

File: src/test/java/org/jenkinsci/plugins/webhookstep/WaitForWebhookTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+import static net.javacrumbs.jsonunit.JsonAssert.assertJsonEquals;
+import static net.javacrumbs.jsonunit.JsonAssert.when;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+import static org.junit.Assert.assertTrue;
+
+public class WaitForWebhookTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/webhookstep/WaitForWebhookTest.java]: I thought you had something using `RestartableJenkinsRule`, or is that coming separately?

**Comment by slide** [comment]
@jglick the restartable test is coming separate. Wanted to land this first

**Pull request merged**